Master

leonfagan71 opened the pull request:
Added {message} to all command actions and alert

leonfagan71 committed:
Merge remote-tracking branch 'refs/remotes/zedwick/master'

leonfagan71 committed:
Added more to commands.

UNTESTED - Didn't have java IDE installed on new PC.

added more to commands, allows {message} and {player} to work on more
actions

leonfagan71 committed:
added message to commands.

added {message} to scommand, pcommand, Ccommand and alert actions.

leonfagan71 committed:
Added message

added {message} to actions message and kick as well as last changes to
add it into commands.

zedwick commented:
There's a lot of code repetition going on there. I had considered requesting you add the same to the other commands, but I couldn't think of a use case for the player sending a command which included their original message, for example. I decided it wasn't really needed.

However, if you'd like to make the placeholders available universally in all actions, we should have the logic in a single place so we don't have to maintain all these different lines of code to do the same thing.

leonfagan71 committed:
added word match

leonfagan71 commented:
I agree, I'll probably make a function for it in a moment I'm just working
on a {match} which states the first word that is matched.

On Sat, Apr 8, 2017 at 12:07 PM, zedwick <notifications@github.com> wrote:

> There's a lot of code repetition going on there. I had considered
> requesting you add the same to the other commands, but I couldn't think of
> a use case for the player sending a command which included their original
> message, for example. I decided it wasn't really needed.
>
> However, if you'd like to make the placeholders available universally in
> all actions, we should have the logic in a single place so we don't have to
> maintain all these different lines of code to do the same thing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zedwick/BungeeChatFilter/pull/11#issuecomment-292711046>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG4O2NHU2SqfLptlv4XHSDICR24lA6B-ks5rt2pagaJpZM4M3qay>
> .
>

leonfagan71 committed:
added function for params

added function for params

leonfagan71 committed:
added customParams function to make configuring params easier.

added customParams function to make configuring params easier.

leonfagan71 commented:
Made it a function, simpler to change things.
{match} = word that matched to cause issue.
{message} = will display the message or command used
{player} = will display the playername

Example:
message: "{player}, you said '{match}' when saying {message}"

zedwick reviewed (changes requested):
You seem to have duplicated config.yml and plugin.yml into the root, could you remove those please.

leonfagan71 committed:
Removing config from root

removing config from root, but keeping plugin as the plugin doesn't run
unless that is present

leonfagan71 committed:
Update .gitignore

leonfagan71 committed:
Delete plugin.yml

leonfagan71 commented:
I've removed plugin.yml and config.yml from the root, 
plugin.yml does cause an issue for me and requires it to be there, I've updated .gitignore for this to not be noticed in the future.

zedwick commented:
Which IDE do you use? If you can open the project as a maven project, the pom.xml should provide the proper location for the plugin.yml and config.yml.

I think I'd prefer it if you didn't add little workarounds like that to the .gitignore. You can always add it to your local .gitignore if you want to still do that.

leonfagan71 commented:
Eclipse Java EE IDE.
For some reason it's ignoring the pom.xml file or it does not specify it in
it.
I will attempt to remove the global .gitignore and just use it locally.

Regards.

On Sat, Apr 8, 2017 at 2:30 PM, zedwick <notifications@github.com> wrote:

> Which IDE do you use? If you can open the project as a maven project, the
> pom.xml should provide the proper location for the plugin.yml and
> config.yml.
>
> I think I'd prefer it if you didn't add little workarounds like that to
> the .gitignore. You can always add it to your local .gitignore if you want
> to still do that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zedwick/BungeeChatFilter/pull/11#issuecomment-292718220>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG4O2J4XUfvhkPIvPFuMl1gRhkEMAdtEks5rt4vwgaJpZM4M3qay>
> .
>

leonfagan71 committed:
Revert "Update .gitignore"

This reverts commit 18867e3b6a456780e9b990bf408e1acc8d2e0b4f.

leonfagan71 commented:
Ahh, It looks like I was exporting instead of using Maven,
Issues resolved now.

zedwick commented on .gitignore:
```diff
@@ -5,4 +5,5 @@ target/
-.git/
\ No newline at end of file
+.git/
+plugin.yml
```
Remove plugin.yml line

zedwick commented on src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:
```diff
@@ -120,4 +132,23 @@ public boolean needsPermission(){
+    	Pattern pattern = Pattern.compile(regex);
+    	Matcher m = pattern.matcher(message);
+    	m.find();
+    	return m.group(1) ;
```
remove space here

zedwick commented on src/main/resources/plugin.yml:
```diff
@@ -1,5 +0,0 @@
-main : com.minecraftdimensions.bungeechatfilter.Main
-version: ${project.version}b-${build.number}
-author: Bloodsplat, Zedwick
-website: https://www.spigotmc.org/resources/bungeechatfilter.20596/
```
Don't remove plugin.yml file.

zedwick commented on src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:
```diff
@@ -53,6 +51,9 @@ public boolean doesMessageContainRegex( String message ) {
         String message = event.getMessage();
+        //get message word that matched.
+        
+        
```
Remove unused comment/line

zedwick commented on src/main/java/com/minecraftdimensions/bungeechatfilter/Rule.java:
```diff
@@ -63,21 +64,32 @@ public void performActions( ChatEvent event, ProxiedPlayer player ) {
+                String rp=customParams(actions.get( action )[0], player, event);
                 if(message.split( " ", 2 ).length>1){
-                       alert =alert.replace("{arguments}", message.split( " ", 2 )[1] )    ;
+                	rp =rp.replace("{arguments}", message.split( " ", 2 )[1] )    ;
```
I know it was there before, but could you remove the whitespace before the `;`

zedwick reviewed (changes requested):
Got some changes for you to make before I can accept this, but once they're done should be ready.

leonfagan71 committed:
Changes Implemented.

leonfagan71 commented:
Have made those changes, any more to recommend?

zedwick reviewed (changes requested):
Could you please undo the changes to the whitespace at the beginning of all lines in Rule.java. The rest of the project uses spaces, where-as you have switched the lines in this file to tabs.

leonfagan71 committed:
Replaced tabs with spaces

leonfagan71 commented:
Like that? 
all tabs replaced with spaces.
chucked it through a few syntax fixers to sort indentations and remove whitespace, that's why tabs existed.

zedwick commented:
Right, I see.
Now I can see that a lot of the other original whitespace has been removed, the project has spaces inserted for readability (I assumed) throughout. I'm not necessarily completely opposed to that changing, but it shouldn't be part of this PR and should be applied throughout the project.

It also makes it difficult to easily see which lines of code has been changed, as I am seeing a lot of other unrelated lines changed due to the spaces being removed.

leonfagan71 commented:
Okay, I understand, 
I'm fairly new to GitHub in terms of actually adding to repositories,
I'll see if I can go back a bit.

zedwick commented:
No worries.
